sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2Suite.scala

```diff
@@ -435,6 +436,27 @@ class DataSourceV2Suite extends QueryTest with SharedSparkSession with AdaptiveS
       }
     }
   }
+
+  Seq("true", "false").foreach { codegenEnabled =>
+    test("SPARK-35287: project generating unsafe row " +
+      s"should not be removed (codegen=$codegenEnabled)") {
```

Comment:
We can move the `codegenEnabled` config into the test itself and move this test to `RemoveRedundantProjectsSuite`.

Reply:
Thanks I'll move this test to `RemoveRedundantProjectsSuite` but `codegenEnabled` outside the test definition is intentional. We can easily know which case (codegen enabled or disabled) fails.
`DataFrameAggregateSuite` does the same way.

https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala#L1005-L1011